packages/clerk-js/src/core/clerk.ts

```diff
@@ -1324,6 +1312,24 @@ export default class Clerk implements ClerkInterface {
     });
   };
 
+  #buildSignInOrUpUrl = (key: 'signInUrl' | 'signUpUrl', options?: RedirectOptions): string => {
+    const opts: RedirectOptions = {
+      afterSignInUrl: extractAuthPathByPriority('afterSignInUrl', { ctx: options, options: this.#options }, false),
```

Comment:
🔧 We should consider moving these to a redirection service that computes these values once and then has getters that are used to get the computed values.

Reply:
This is a fair point, although the available values are different depending on where this redirection service is called. For example, inside the `useSignInContext` hook, we can access query params, SignIn component props, ClerkProvider-level props and displayConfig. On the other hand, in `clerk.ts` we only have access to ClerkProvider props, displayConfig and maybe `RedirectOptions` from some components. So the computation maybe cannot be done once for all.

I can document where we use each of these, so that we can come up with a better solution that centralizes the whole logic for redirections.